test/test_other.py

```diff
@@ -10532,6 +10544,28 @@ def test_assertions_on_outgoing_module_api_changes(self):
     self.do_runf('src.cpp', expected, emcc_args=['-sASSERTIONS'])
 
   def test_assertions_on_ready_promise(self):
+    # check that when assertions are on we give useful error messages for
+    # mistakenly thinking the Promise is an instance. I.e., once you could do
+    # Module()._main to get an instance and the main function, but after
+    # the breaking change in #10697 Module() now returns a promise, and to get
+    # the instance you must use .then() to get a callback with the instance.
+    create_file('test.js', r'''
+      try {
+        Module._main;
+      } catch(e) {
+        console.log(e);
+      }
+      try {
+      } catch(e) {
+        console.log(e);
+      }
+    ''')
+    self.run_process([EMCC, test_file('hello_world.c'), '-sASSERTIONS', '--extern-post-js', 'test.js'])
+    # A return code of 7 is from the unhandled Promise rejection
+    out = self.run_js('a.out.js', assert_returncode=0)
+    self.assertContained('hello, world!', out)
```

Comment:
I don't understand this test.. its doesn't seem to use MODULARIZE at all.. and it checking for 0 return code..  are the comments out of date?

Reply:
Indeed not necessary - I just shouldn't work on PRs at midnight ...

Reply:
Indeed not necessary - I just shouldn't work on PRs at midnight ...